orchestrator/orchestrator/src/main_loop.rs

```diff
@@ -59,31 +58,36 @@ pub async fn orchestrator_main_loop(
     blocks_to_search: u64,
     gas_adjustment: f64,
     relayer_opt_out: bool,
-    cosmos_msg_batch_size: u32,
-) {
-    let (tx, rx) = tokio::sync::mpsc::channel(1);
+    cosmos_msg_batch_size: usize,
+) -> Result<(), GravityError> {
+    let cosmos_address = cosmos_key.to_address(&contact.get_prefix())?;
+    let block_delay = get_block_delay(eth_client.clone()).await?;
+    let gravity_id = get_gravity_id(gravity_contract_address, eth_client.clone()).await?;
+    let (tx, rx) = tokio::sync::mpsc::channel(10);
 
     let a = send_main_loop(
-        &contact,
+        contact.clone(),
```

Comment:
What is the benefit of cloning `contact` rather than using the same instance as it was doing before? Is there a concern around multiple threads using it and account sequencing issues? I do wonder if cloning the object isn't just going to create multiple versions of e.g. previous account sequence state, rather than recreating a new `Contact` when it's needed.

Reply:
The contact object doesn't maintain account state, but actually queries the required tx args every time it builds a transaction. See `send_messages`, specifically the call to `.get_message_args(...)` is where this happens.

Reply:
Sorry, in regards to the real purpose of your question: it's another borrow checker/lifetime thing.